src/cider/nrepl/middleware/util/instrument.clj

```diff
@@ -253,50 +267,74 @@
   ;; Don't use `postwalk` because it destroys previous metadata.
   (walk-indexed #(tag-form %2 breakfunction) form))
 
+(defn print-form
+  "Pretty print form.
+  If expand? is true, macroexpand the form. If meta? is true, also print
+  meta. This function is intended for inspection of instrumented code."
+  [form & [expand? meta?]]
+  (binding [*print-meta* meta?]
+    (let [form (if expand?
+                 (m/macroexpand-all form)
+                 form)]
+      (clojure.pprint/pprint form)))
+  (flush)
+  form)
+
+(defn- strip-instrumentation-meta
+  "Remove all tags in order to reduce java bytecode size and enjoy cleaner code
+  printouts. We keep ::breakfunction for def symbols because that is how we
+  identify currently instrumented vars in list-instrumented-defs."
+  [form]
+  (walk-indexed
+    (fn [_ f]
+      (if (instance? clojure.lang.IObj f)
+        (let [keys [::original-form ::extras ::breakfunction ::def-symbol]
+              f    (if (::def-symbol (meta f))
+                     (let [br (::breakfunction (meta f))
+                           f1 (m/strip-meta f keys)]
+                       (if br
+                         (vary-meta f1 assoc :instrumented (name (:name (meta br))))
```

Comment:
Final form has only `:instrumented` string tag in it. Keeping full function around leads to huge clutter when you print the form with meta and generally makes it much harder to understand what is what. Each meta key also adds 5 extra bytecode ops to the final object size on every breakpoint invocation. After all it's just a tag and should be a string or a symbol, especially when you send it to emacs.

The non-qualified `:instruented` is intentional. I am not bound to it but it looks to me that qualification is not warranted in this context. The`:instrumented` tag is more generic than `:cider.nrepl.middleware.util.instrument/instrumented`. Different middleware or different implementation of instrumentation should be able to use it without sending that sausage around. It also rather annoying in printouts as doesn't bear extra informational content there. On emacs  side it feels like spillage of internals as well. See [associated changes](https://github.com/clojure-emacs/cider/pull/2002).